packages/integration-objects/srv/validations/Validations.js

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+const util = require('@sap-cloud-sdk/util');
+const { ValidationError } = require('../exceptions/validation-error');
+const ExtensionConstants = require('../utils/ExtensionConstants');
+const ErrorStatuses = require('../utils/ErrorStatuses');
+
+// validate the tenantid in the request payload
+function validateTenantIdInPayload(req) {
+  const tId = req.user.tenant;
+  if (!util.isNullish(req.data.tenantID) && tId !== req.data.tenantID) {
+    throw new ValidationError(ExtensionConstants.INVALID_TENANTID, ErrorStatuses.BAD_REQUEST);
+  } else if (util.isNullish(req.data.tenantID)) {
+    req.data.tenantID = tId;
```

Comment:
This is very dangerous. The method is called validate but it sets a value which is a side effect you do not expect form the method name. If this is necessary split it in a validate method and a set method or keep it as a single method and call it set.

Reply:
Have fixed this and added a different method called setTenantId